src/averageAges.js

```diff
@@ -49,13 +76,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * If `onlyWithSon` is specified then function calculates age difference only
  * for sons and their mothers.
  *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
+//  * @param {object[]} people
+//  * @param {boolean} onlyWithSon - optional
+//  *
+//  * @return {number}
+//  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const differences = [];
+
+  people.forEach(person => {
+    const personChildren = onlyWithSon
+      ? people
+        .filter(child => person.name === child.mother && child.sex === 'm')
+      : people
+        .filter(child => person.name === child.mother);
```

Comment:
You use a filter in both cases. Maybe it's better to do a check `onlyWithSon` condition inside the filter?